File: app_main/src/main/java/app/jw/mapable/gm/Start/SplashActivity.java
package app.jw.mapable.gm.Start;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import app.jw.mapable.gm.FirstSetting.FirstSettingActivity;
import app.jw.mapable.gm.R;

public class SplashActivity extends AppCompatActivity {

    SharedPreferences preferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        splashAnimation();

    }

    private void splashAnimation()
    {
        ImageView imageView = findViewById(R.id.imageView);
        ImageView imageView1 = findViewById(R.id.imageView2);
        TextView textView = findViewById(R.id.textView3);
        TextView textView1 = findViewById(R.id.textView10);
        Animation textAnim = AnimationUtils.loadAnimation(this, R.anim.anim_fade_in);
        textView.startAnimation(textAnim);
        textView1.startAnimation(textAnim);
        Animation imageAnim = AnimationUtils.loadAnimation(this, R.anim.anim_rotate);
        imageView.startAnimation(imageAnim);
        Animation newImageAnim = AnimationUtils.loadAnimation(this, R.anim.anim_zoom_in);
        ImageView newImageView = findViewById(R.id.imageView4);
        ImageView newImageView2 = findViewById(R.id.imageView5);
        newImageView2.startAnimation(newImageAnim);
        newImageView.startAnimation(newImageAnim);


        newImageAnim.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

                newImageView.setVisibility(View.GONE);
                newImageView2.setVisibility(View.GONE);

                Handler handler = new Handler(); // Handler 사용해 1초가 지나면 MainActivity 열기
                handler.postDelayed(() -> {
                    textView.setVisibility(View.GONE);
                    textView1.setVisibility(View.GONE);
                    imageView1.setVisibility(View.GONE);
                    imageView.setVisibility(View.GONE);
                    newImageView.setVisibility(View.VISIBLE);
                    newImageView2.setVisibility(View.VISIBLE);
                }, 2300);

                Handler handler2 = new Handler(); // Handler 사용해 1초가 지나면 MainActivity 열기
                handler2.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        preferences = getSharedPreferences("preference", 0);
                        if(preferences.getBoolean("settingAvailable", false))
                        {
                            Intent intent = new Intent(SplashActivity.this, StartActivity.class);
                            startActivity(intent);

                            finish();
                            overridePendingTransition(R.anim.anim_none, R.anim.anim_zoom_in);
                        }
                        else
                        {
                            Intent intent = new Intent(SplashActivity.this, FirstSettingActivity.class);
                            startActivity(intent);
                            finish();
                            overridePendingTransition(R.anim.anim_none, R.anim.anim_none);
                        }
                    }
                }, 3700);

            }

            @Override
            public void onAnimationEnd(Animation animation) {

               // overridePendingTransition(R.anim.anim_slide_out_top, R.anim.anim_slide_in_down);






            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });

    }

    private final long backKeyPressedTime = 0;
    // 첫 번째 뒤로 가기 버튼을 누를 때 표시
    private Toast toast;

//    @Override
//    public void onBackPressed() {
//
//        if (System.currentTimeMillis() > backKeyPressedTime + 2500) {
//            backKeyPressedTime = System.currentTimeMillis();
//            toast = Toast.makeText(this, "뒤로 가기 버튼을 한 번 더 누르시면 종료됩니다.", Toast.LENGTH_LONG);
//            toast.show();
//            return;
//        }
//        // 마지막으로 뒤로 가기 버튼을 눌렀던 시간에 2.5초를 더해 현재 시간과 비교 후
//        // 마지막으로 뒤로 가기 버튼을 눌렀던 시간이 2.5초가 지나지 않았으면 종료
//        if (System.currentTimeMillis() <= backKeyPressedTime + 2500) {
//            android.os.Process.killProcess(android.os.Process.myPid());
//            finish();
//        }
//
//
//    }
}

File: app_main/src/main/java/app/jw/mapable/gm/Splash/SplashActivity.kt
package app.jw.mapable.gm.Splash

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import app.jw.mapable.gm.FirstSetting.FirstSettingActivity
import app.jw.mapable.gm.R
import app.jw.mapable.gm.Start.StartActivity
import kotlinx.android.synthetic.main.activity_after_search.*
import kotlinx.android.synthetic.main.activity_splash.*
import kotlinx.android.synthetic.main.dialog_start.*

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        splashAnimation()

        val preferences : SharedPreferences = getSharedPreferences("preferences", 0)
        preferences.edit().putBoolean("start", false).apply()
        preferences.edit().putBoolean("end", false).apply()


    }

    fun splashAnimation()
    {

        val textAnim : Animation = AnimationUtils.loadAnimation(this, R.anim.anim_fade_in);
        textView3.startAnimation(textAnim)
        textView10.startAnimation(textAnim)
        val imageAnim = AnimationUtils.loadAnimation(this, R.anim.anim_rotate)
        imageView.startAnimation(imageAnim)
        val newImageAnim = AnimationUtils.loadAnimation(this, R.anim.anim_zoom_in)

        imageView5.startAnimation(newImageAnim)
        imageView4.startAnimation(newImageAnim)

        newImageAnim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {
                imageView4.visibility = View.GONE
                imageView5.visibility = View.GONE

                val handler = Handler()

                handler.postDelayed({
                    textView3.visibility = View.GONE
                    textView10.visibility = View.GONE
                    imageView.visibility = View.GONE
                    imageView2.visibility = View.GONE

                    imageView4.visibility = View.VISIBLE
                    imageView5.visibility = View.VISIBLE


                }, 2300)

                val handler2 = Handler()

                //TODO : 애니메이션 오류로 Zoom In 비활성화
                handler2.postDelayed({

                    val preferences : SharedPreferences = getSharedPreferences("preferences", 0)
                    if (preferences.getBoolean("settingAvailable", false)) {
                        val intent = Intent(this@SplashActivity, StartActivity::class.java)
                        startActivity(intent)
                        finish()
                        overridePendingTransition(R.anim.anim_none, R.anim.anim_zoom_in)
                    } else {
                        val intent = Intent(this@SplashActivity, FirstSettingActivity::class.java)
                        startActivity(intent)
                        finish()
                        overridePendingTransition(R.anim.anim_none, R.anim.anim_none)
                    }

                }, 2300)



            }

            override fun onAnimationEnd(animation: Animation) {

            }

            override fun onAnimationRepeat(animation: Animation) {

            }
        })

    }
}


